source/common/network/dns_impl.cc

```diff
@@ -215,11 +219,21 @@ ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name,
   }
 }
 
-void DnsResolverImpl::PendingResolution::getHostByName(int family) {
-  ares_gethostbyname(
-      channel_, dns_name_.c_str(), family,
-      [](void* arg, int status, int timeouts, hostent* hostent) {
-        static_cast<PendingResolution*>(arg)->onAresHostCallback(status, timeouts, hostent);
+void DnsResolverImpl::PendingResolution::getAddrInfo(int family) {
+  struct ares_addrinfo_hints hints = {};
+  hints.ai_family = family;
+
+  /**
+   * ARES_AI_ENVHOSTS read hosts file path from the environment variable
```

Comment:
Sorry for the late review but do we want this? Is this what was done previously? If the hosts file can be overriden via an env variable should we document that?

Reply:
Yeah, maybe just drop this one. It seemed harmless, it just pays attention to `CARES_HOSTS`, but it's not really needed.

Reply:
It seems to be harmless as well as another one. I will drop both of them.

Reply:
There seems a bug in `ares__sortaddrinfo.c`. I will add `ARES_AI_NOSORT` back.

Reply:
Yeah, we definitely want `NOSORT` I believe, to avoid modifying DNS results and having c-ares do additional network stuff.